### PR TITLE
MM-49868: revert openid packaging changes

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

This PR brings OpenID auth to 7.5 release.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-49868
#### Release Note

```release-note
Revert cloud license checks for OpenID authentication methods.
```

### CODE REVIEW

File: config/client.go [hunk 1]
```diff
@@ -333,22 +333,6 @@ func GenerateLimitedClientConfig(c *model.Config, telemetryID string, license *m
...
-		}
-
-		if *license.Features.MFA {
-			props["EnforceMultifactorAuthentication"] = strconv.FormatBool(*c.ServiceSettings.EnforceMultifactorAuthentication)
-		}
-
-		if license.IsCloud() {
-			// MM-48727: enable SSO options for free cloud - not in self hosted
-			*license.Features.GoogleOAuth = true
-			*license.Features.Office365OAuth = true
```
Comment (mkraft) [inline config/client.go:348]: Will this change disable Google and Office 365 authentication in Cloud Free?

Comment (plant99) [reply config/client.go:348]: If their license supports Google, O365, which is valid in case of Pro / Ent licenses, they can use this feature.

Comment (BenCookie95) [reply config/client.go:348]: I'm a little confused, doesn't the [ticket](https://mattermost.atlassian.net/browse/MM-49868) indicate that cloud free should have Google and Office 365 enabled?

**Review by BenCookie95 — CHANGES_REQUESTED** [review]

**Comment by mkraft** [comment]
FYI this PR is still showing Draft status.

**Pull request closed without merging**